cpp/include/raft/matrix/detail/linewise_op.cuh

```diff
@@ -176,6 +186,37 @@ struct Linewise {
       return out;
     }
   }
+
+  /**
+   * @brief Same as loadVec, but padds data with Ones
+   *
+   * @param shm
+   * @param p
+   * @param blockOffset
+   * @param rowLen
+   * @param rowLenPadded
+   * @return a contiguous chunk of a vector, suitable for `vectorRows`.
+   */
+  static __device__ __forceinline__ Vec loadVecPadded(Type* shm,
+                                                      const Type* p,
+                                                      const IdxType blockOffset,
+                                                      const IdxType rowLen,
+                                                      const IdxType rowLenPadded) noexcept
+  {
+    IdxType j = blockOffset + threadIdx.x;
+#pragma unroll VecElems
+    for (int k = threadIdx.x; k < VecElems * BlockSize; k += BlockSize, j += BlockSize) {
+      while (j >= rowLenPadded)
+        j -= rowLenPadded;
+      shm[k] = j < rowLen ? p[j] : Type(1);
```

Comment:
Out of curiosity: why ones and not zeroes? :)

Reply:
I had division ops in mind and did not want to risk any division by zero. I was unsure whether this might cause issues with tools like valgrind?

Reply:
Hmm, good idea, I guess. I don't think it will cost anything anyway.